packages/ember-runtime/lib/system/each_proxy.js

```diff
@@ -104,91 +18,55 @@ function removeObserverForContentKey(content, keyName, proxy, idx, loc) {
   @class EachProxy
   @private
 */
-var EachProxy = EmberObject.extend({
-
-  init(content) {
-    this._super(...arguments);
-    this._content = content;
-    content.addArrayObserver(this);
-
-    // in case someone is already observing some keys make sure they are
-    // added
-    watchedEvents(this).forEach((eventName) => {
-      this.didAddListener(eventName);
-    });
-  },
-
-  /**
-    You can directly access mapped properties by simply requesting them.
-    The `unknownProperty` handler will generate an EachArray of each item.
-
-    @method unknownProperty
-    @param keyName {String}
-    @param value {*}
-    @private
-  */
-  unknownProperty(keyName, value) {
-    var ret = new EachArray(this._content, keyName, this);
-    defineProperty(this, keyName, null, ret);
-    this.beginObservingContentKey(keyName);
-    return ret;
-  },
+function EachProxy(content) {
+  this._content = content;
+  this._keys = undefined;
+  this.__ember_meta__ = undefined;
+}
 
+EachProxy.prototype = {
   // ..........................................................
   // ARRAY CHANGES
   // Invokes whenever the content array itself changes.
 
   arrayWillChange(content, idx, removedCnt, addedCnt) {
-    var keys = this._keys;
-    var key, lim;
-
-    lim = removedCnt>0 ? idx+removedCnt : -1;
-    beginPropertyChanges(this);
-
-    for (key in keys) {
-      if (!keys.hasOwnProperty(key)) { continue; }
-
-      if (lim>0) { removeObserverForContentKey(content, key, this, idx, lim); }
-
+    let keys = this._keys;
```

Comment:
is `this._keys` ever legitimately contain prototype extensions? If not, maybe we should make it an `EmptyObject`. This would allow us to also skip the bellow `!keys.hasOwnProperty(key)`